pkg/pillar/cmd/domainmgr/containerd.go

```diff
@@ -36,44 +39,45 @@ type KeyValue struct {
 
 // getContainerPath return the path to the root of the container. This is *not*
 // necessarily the rootfs, which may be a layer below
-func getContainerPath(id string) string {
-	return path.Join(containersRoot, id)
+func getContainerPath(containerID string) string {
+	return path.Join(containersRoot, CtrdServicesNamespace, containerID)
 }
 
 // getContainerRootfs return the path to the root of the container filesystem
-func getContainerRootfs(id string) string {
-	return path.Join(getContainerPath(id), containerRootfsPath)
+func getContainerRootfs(containerID string) string {
+	return path.Join(getContainerPath(containerID), containerRootfsPath)
 }
 
 // containerdLoadImageTar load an image tar into the containerd content store
-func containerdLoadImageTar(filename string) (map[string]string, error) {
+func containerdLoadImageTar(filename string) ([]string, error) {
 	// load the content into the containerd content store
-	ctr, err := containerd.New(ctrdSocket)
+	var err error
+
+	ctrdClient, err := getContainerdClient(ctrdSocket)
 	if err != nil {
-		log.Errorf("could not connect to containerd socket at %s: %+s", ctrdSocket, err.Error())
-		return nil, err
+		log.Errorf("Could not build containerd client. %v", err.Error())
+		return nil, fmt.Errorf("Exception while loading container. %v", err)
 	}
-	defer ctr.Close()
-
-	ctrdCtx := namespaces.WithNamespace(context.Background(), CtrdServicesNamespace)
+	defer ctrdClient.Close()
+	ctrdCtx := getContainerdContext(CtrdServicesNamespace)
 
 	tarReader, err := os.Open(filename)
 	if err != nil {
 		log.Errorf("could not open tar file for reading at %s: %+s", filename, err.Error())
 		return nil, err
 	}
 
-	imgs, err := ctr.Import(ctrdCtx, tarReader)
+	imgs, err := ctrdClient.Import(ctrdCtx, tarReader)
 	if err != nil {
 		log.Errorf("could not load image tar at %s into containerd: %+s", filename, err.Error())
 		return nil, err
 	}
 	// successful, so return the list of images we imported
-	names := make(map[string]string)
+	imageNames := make([]string, 0)
 	for _, tag := range imgs {
-		names[tag.Name] = string(tag.Target.Digest)
+		imageNames = append(imageNames, tag.Name)
 	}
-	return names, nil
+	return imageNames, nil
```

Comment:
Why convert this from a `map[string]string` with the hashes to just a `[]string`? We lose some information that way. True, we don't use it yet, but why lose the info?

Reply:
NewContainer() requires image as containerd.Image{} type, but ImageImport() returned image as images.Image{} type. So to convert that, I did getImage() with the imageName which returned the image as containerd.Image{} type. 
But as you mentioned, this is clumsy as the same image is fetched twice. So on digging into container code for a simpler way to convert that.

Reply:
Got it. This one didn't matter all that much, but nice to see